src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -53,13 +52,37 @@ public IslandManager() {
         super(IridiumSkyblock.getInstance());
     }
 
+    public boolean deleteWorld(File path) {
+        if(path.exists()) {
+            File files[] = path.listFiles();
+            for(int i=0; i<files.length; i++) {
+                if(files[i].isDirectory()) {
+                    deleteWorld(files[i]);
+                } else {
+                    files[i].delete();
+                }
+            }
+        }
+        return(path.delete());
+    }
+
+    // When creating the cached world, we specify its name with a path to place it in the data folder.
+    // For some silly reason, using getDataFolder() results in an illegal character error.
+    // Same with using File.separator.
+    public String getCacheWorldName(World world) {
+        return "plugins/iridiumskyblock" + "/" + "regenWorlds" + "/" + world.getName() + "_regen";
```

Comment:
Use `JavaPlugin#getDataFolder()` instead

Reply:
Just read the comment, lmao. What exactly is the stack trace?

Reply:
I know that Paper is considering moving configs to a different folder in the long term, so something like this is less than optimal

Reply:
```console
[14:09:06 WARN]: [IridiumSkyblock] Task #38 for IridiumSkyblock v4.0.9-OCEAN generated an exception
java.lang.IllegalArgumentException: Invalid key. Must be [a-z0-9/._-]:
```
i would definitely prefer using
```java
IridiumSkyblock.getInstance().getDataFolder().getName() + File.separator + "regenWorlds" + File.separator + world.getName() + "_regen"
```
but uh yeah its bein dumb